packages/utils/src/types/unacknowledged.ts

```diff
@@ -14,20 +13,24 @@ export class UnacknowledgedTicket {
   public serialize(): Uint8Array {
     return serializeToU8a([
       [this.ticket.serialize(), Ticket.SIZE],
-      [this.secretA.serialize(), Hash.SIZE]
+      [this.ownKey.serialize(), Hash.SIZE]
     ])
   }
 
-  private verifyChallenge(hashedKeyHalf: Uint8Array): boolean {
-    return Hash.create(u8aConcat(this.secretA.serialize(), hashedKeyHalf)).hash().eq(this.ticket.challenge)
+  public verifySignature(signer: PublicKey): boolean {
+    return this.ticket.verify(signer)
   }
 
-  private async verifySignature(peerId: PeerId): Promise<boolean> {
-    return this.ticket.verify(new PublicKey(peerId.pubKey.marshal()))
-  }
+  verify(signer: PublicKey, acknowledgement: Hash): ReturnType<typeof validateAcknowledgement> {
```

Comment:
why is this not a Promise<boolean> any more?

Reply:
Secp256k1 is synchronous, the function has been async because the equivalent in Polkadot.js is async

Reply:
Well we aren't targeting Polkadot - in any case, the return type should be boolean.

Reply:
This is what I'm saying and this is the reason why it is no longer async. Also changed the return type